### DISCUSSION TITLE
Feature request: allow for disabling of Bindings

### DISCUSSION THREAD

**Discussion by lolbat** [Textualize/textual#2242] [Ideas]
Currently the bindings in Textual are binary. The `show` attribute just hides them but they are still active.

I would like to be able to have a `disabled` attribute that would be read/write that would show the binding but in a disabled state.

There are instances when you want to have commands active only in a particular context. I don't want to show the Edit command if a file isn't selected and I don't want to show the Write command if the file isn't dirty.

Suggestions
1. Add a read/write disabled attribute
2. Add a Binding--disabled class component
3. Add a Binding--enabled class component

**Comment by liancheng** [comment]
+1

My use case is modal key binding similar to Vim. E.g.:

1. Each focusable widget is assigned to a key binding that is initially disabled.
2. Press `K` enters hotkey mode and enables those key bindings.
3. Activating such key binding moves the focus to the target widget.
